### PR TITLE
Enable directly using AWS S3 with --recipe-url parameter of Chef Client

### PR DESCRIPTION
**Pull request by tecracer-theinen**
## Description

Using `--recipe-url` for quick setups of AWS EC2 instances requires the cookbook TGZ files to be public. With this small patch, you can directly specify an object on S3 like `--recipe-url s3://my-private-bucket/setup-0.1.0.tgz`.

## Related Issue

none

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by lamont-granquist** [comment]
you probably need to `bundle update` or `rake dependencies:update_gemfile_lock` to bump the Gemfile.lock

i can't seem to check out your branch this morning.

**Comment by tecracer-theinen** [comment]
Thanks @lamont-granquist that actually helped with most pipelines :+1: 

The remaining failures are apparently CentOS 7 and fail on building the `x25519` dependency for knife. Is that something related to my PR or a known issue? It looks only knife-related to me :thinking:

**Comment by tas50** [comment]
RHEL 7 is a known CI issue right now

File: chef.gemspec [hunk 1]
```diff
@@ -55,6 +55,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency "proxifier", "~> 1.0"
 
+  s.add_dependency "aws-sdk-s3", "~> 1.91"
```
Comment (tas50) [inline chef.gemspec:58]: can you add a comment inline here saying what feature this enables similar to the hashi comment below

**Comment by tas50** [comment]
This is neat. Thanks @tecracer-theinen

**Pull request merged**